packages/api/endpoints/pdrs.js

```diff
@@ -66,9 +70,12 @@ async function del(req, res) {
   await deleteS3Object(process.env.system_bucket, pdrS3Key);
 
   const pdrModel = new models.Pdr();
+  const pdrPgModel = new PdrPgModel();
+  const knex = await getKnexClient();
 
   try {
     await pdrModel.delete({ pdrName });
+    await pdrPgModel.delete(knex, { name: pdrName });
```

Comment:
Is there any value to having these in a transaction where, if the Dynamo delete fails, the PG delete is rolled back?

Reply:
I was thinking about that, but that would mean the API has no functionality to delete a record if it doesn't exist in both places. That doesn't seem preferable? We could discuss it with the team as I am a relative newbie to the PG side of things.

Reply:
fwiw, I think we need a transaction as Nate described. Though, a DynamoDB delete can't fail if the record isn't found.
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_DeleteItem.html
> Unless you specify conditions, the DeleteItem is an idempotent operation; running it multiple times on the same item or attribute does not result in an error response.

So, I would have a transaction where we try to delete the PDR from PG first, and then attempt to delete from DynamoDB.

Reply:
Another thing to keep in mind is that a "failing" PDR delete in the PG table won't return an error, it seems to return 0 if not successful and 1 otherwise.

Reply:
I think there is some ambiguity in what we're guarding against with our transactions here. I ran into these same questions on Granule deletes. 

The `Knex.del()` method will returned the number of affected rows. So if no records were deleted, you're totally right. We'll just return 0. Otherwise we'll return the number of deleted records:

>Resolves the promise / fulfills the callback with the number of affected rows for the query.

But I think thats ok and not really what we're trying to guard against here? If a transaction didn't delete anything because there wasn't anything to delete, that transactions should be considered successful. I think where we want to fail the transaction is in cases where there's an actual database error. Like if the PG database is unavailable when we tried to reach it.

So in my mind having a transaction is right and the order that @botanical suggests seems spot-on. We try a PG delete and, if that's successful, we try a Dynamo delete. Dynamo deletes cannot be rolled back so doing that second makes sense.